src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -582,5 +655,227 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
 
             return parameters.Count - 1;
         }
+
+        protected (IEnumerable<T> parameters, IEnumerable<SyntaxToken> separators) UpdateDeclarationBase<T>(
+            SeparatedSyntaxList<T> list,
+            SignatureChange updatedSignature,
+            Func<AddedParameter, T> createNewParameterMethod) where T : SyntaxNode
+        {
+            var originalParameters = updatedSignature.OriginalConfiguration.ToListOfParameters();
+            var reorderedParameters = updatedSignature.UpdatedConfiguration.ToListOfParameters();
+
+            int numAddedParameters = 0;
+
+            var newParameters = new List<T>();
+            for (var index = 0; index < reorderedParameters.Count; index++)
+            {
+                var newParam = reorderedParameters[index];
+                if (newParam is ExistingParameter existingParameter)
+                {
+                    var pos = originalParameters.IndexOf(p => p is ExistingParameter ep && ep.Symbol == existingParameter.Symbol);
+                    if (pos >= 0)
+                    {
+                        var param = list[pos];
+
+                        // copy whitespace trivia from original position
+                        param = TransferLeadingWhitespaceTrivia(param, list[index - numAddedParameters]);
+                        newParameters.Add(param);
+                    }
+                }
+                else
+                {
+                    // Added parameter
+                    numAddedParameters++;
+                    var newParameter = createNewParameterMethod(newParam as AddedParameter);
+                    newParameters.Add(newParameter);
+                }
+            }
+
+            int numSeparatorsToSkip;
+            if (originalParameters.Count == 0)
+            {
+                // () 
+                // Adding X parameters, need to add X-1 separators.
+                numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count + 1;
+            }
+            else
+            {
+                // (a,b,c)
+                // Adding X parameters, need to add X separators.
+                numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count;
+            }
+
+            return (newParameters, GetSeparators(list, numSeparatorsToSkip));
+        }
+
+        protected List<SyntaxToken> GetSeparators<T>(SeparatedSyntaxList<T> arguments, int numSeparatorsToSkip = 0) where T : SyntaxNode
```

Comment:
Does anybody call this with the default of numSeparatorsToSkip?

Reply:
No. Default value removed in next review.